applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt_scan_for_broadcast.c

```diff
@@ -0,0 +1,186 @@
+
+	/* Enforce restraints */
+	timeout = CLAMP(timeout, BT_GAP_PER_ADV_MIN_TIMEOUT, BT_GAP_PER_ADV_MAX_TIMEOUT);
+
+	return timeout;
+}
+
+static void periodic_adv_sync(const struct bt_le_scan_recv_info *info, uint32_t broadcast_id)
+{
+	int ret;
+	struct bt_le_per_adv_sync_param param;
+	struct bt_le_per_adv_sync *pa_sync;
+
+	struct bt_mgmt_msg msg;
+
+	bt_le_scan_cb_unregister(&scan_callback);
+	cb_registered = false;
+
+	ret = bt_le_scan_stop();
+	if (ret) {
+		LOG_WRN("Stop scan failed: %d", ret);
+	}
+
+	bt_addr_le_copy(&param.addr, info->addr);
+	param.options = 0;
+	param.sid = info->sid;
+	param.skip = PA_SYNC_SKIP;
+	param.timeout = interval_to_sync_timeout(info->interval);
+
+	ret = bt_le_per_adv_sync_create(&param, &pa_sync);
+	if (ret) {
+		LOG_ERR("Could not sync to PA: %d", ret);
+		return;
+	}
+
+	msg.event = BT_MGMT_PA_SYNCED;
+	msg.broadcast_id = broadcast_id;
+	msg.pa_sync = pa_sync;
+
+	ret = zbus_chan_pub(&bt_mgmt_chan, &msg, K_NO_WAIT);
+	ERR_CHK(ret);
+}
+
+/**
+ * @brief	Check and parse advertising data for broadcast name and ID
+ *
+ * @param	data		Advertising data to check and parse
+ * @param	user_data	Will contain pointer to broadcast_source struct to be populated
+ *
+ * @return	Will always return true to make sure we find all relevant data
+ */
+static bool scan_check_broadcast_source(struct bt_data *data, void *user_data)
+{
+	struct broadcast_source *source = (struct broadcast_source *)user_data;
+	struct bt_uuid_16 adv_uuid;
+
+	if (data->type == BT_DATA_BROADCAST_NAME && data->data_len) {
+		if (data->data_len <= source->size) {
+			memcpy(source->name, data->data, data->data_len);
+		}
+		return true;
+	}
+
+	if (data->type != BT_DATA_SVC_DATA16) {
+		return true;
+	}
+
+	if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
+		return true;
+	}
+
+	if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
+		return true;
+	}
+
+	if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO)) {
+		return true;
+	}
+
+	source->broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+
+	/* Stop parsing */
+	return true;
+}
+
+/**
+ * @brief      Scan receive callback when scanning for broadcasters
+ *
+ * @param[in]  info  LE advertisement and scan response packet information
+ * @param      ad  Received advertising data
+ */
+static void scan_recv_cb(const struct bt_le_scan_recv_info *info, struct net_buf_simple *ad)
+{
+	char name[BLE_SEARCH_NAME_MAX_LEN] = {'\0'};
+	struct broadcast_source source = {name, ARRAY_SIZE(name), INVALID_BROADCAST_ID};
+
+	/* We are only interested in non-connectable periodic advertisers */
+	if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) || info->interval == 0) {
+		return;
+	}
+
+	bt_data_parse(ad, scan_check_broadcast_source, (void *)&source);
+
+	if (source.broadcast_id != INVALID_BROADCAST_ID) {
+		if (strlen(source.name) == strlen(srch_name)) {
+			if (strncmp(source.name, srch_name, strlen(srch_name)) == 0) {
```

Comment:
Wouldn't it be more efficient to throw away all broadcasters that doesn't match in length before trying to compare the name? I do agree with using strcmp rather than strncmp though.

Reply:
Removed, and will use strncmp